### PR TITLE
Add fenced frames BCD

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

Fenced frames are part of the set of [Privacy Sandbox](https://developer.chrome.com/docs/privacy-sandbox/) features being released by default in Chrome 115 and friends. This PR aims to add BCD for all the new features that the [fenced frame](https://wicg.github.io/fenced-frame/) spec brings.

See my [research document](https://docs.google.com/document/d/1r27C2_yV8jb7N4-7MxLMHWCx9wHHaCw9HbBDcTyPnZs/edit) for details of all the features added by this PR.

### CODE REVIEW

**Comment by queengooborg** [comment]
Based upon results from the mdn-bcd-collector project, it appears that this set of features has landed in Chrome 117.  May we update the data accordingly?

**Comment by chrisdavidmills** [comment]
> Based upon results from the mdn-bcd-collector project, it appears that this set of features has landed in Chrome 117. May we update the data accordingly?

@queengooborg done.

File: http/headers/Content-Security-Policy.json [hunk 1]
```diff
@@ -295,6 +295,40 @@
             }
           }
         },
+        "fenced-frame-src": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Security-Policy/fenced-frame-src",
+            "spec_url": "https://wicg.github.io/fenced-frame/#new-csp-directive",
+            "support": {
+              "chrome": {
+                "version_added": "115"
```
Comment (queengooborg) [inline http/headers/Content-Security-Policy.json:304]:
```suggestion
                "version_added": "117"
```

**Review by queengooborg — CHANGES_REQUESTED** [review]

**Review by Elchi3 — COMMENTED** [review]
Looking at the IDL, are these features missing in this PR or do they not have an implementation?
- api.Navigator.deprecatedReplaceInURN
- api.HTMLFencedFrameElement.HTMLFencedFrameElement (constructor)

**Comment by chrisdavidmills** [comment]
> * api.Navigator.deprecatedReplaceInURN
> 
> * api.HTMLFencedFrameElement.HTMLFencedFrameElement (constructor)

@Elchi3 Hrm, these weren't in the spec when I first started documenting this (it was a while ago). I've had a look in the Chromium source code, and they are there in the IDL.

I've dropped the Chrome eng folk an email to ask when these items were first supported.

**Comment by Elchi3** [comment]
Marking this as draft, please set as ready once you want me to re-review.

**Comment by chrisdavidmills** [comment]
> Marking this as draft, please set as ready once you want me to re-review.

@Elchi3 OK, I've looked into the missing items some more, and have come to the following conclusions:

- `api.Navigator.deprecatedReplaceInURN`: This is definitely supported, and was introduced at the same time as the rest of the fenced frames features. I've added this here, and will also add a page to the corresponding content PR.
- `api.HTMLFencedFrameElement.HTMLFencedFrameElement (constructor)`: I don't know where I got that this was included in Chrome from. I later found [this IDL](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/fenced_frame/html_fenced_frame_element.idl), which doesn't include it, and when you try to use it in JS, you get an "Illegal constructor" error. So I am going to leave it out for now.

**Comment by Elchi3** [comment]
> * `api.HTMLFencedFrameElement.HTMLFencedFrameElement (constructor)`: I don't know where I got that this was included in Chrome from. I later found [this IDL](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/fenced_frame/html_fenced_frame_element.idl), which doesn't include it, and when you try to use it in JS, you get an "Illegal constructor" error. So I am going to leave it out for now.

Thanks for looking into this, Chris! I also just now realized that the spec uses an "HTMLConstructor". I will need to read into what this means exactly. See https://html.spec.whatwg.org/multipage/dom.html#html-element-constructors.

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks Chris. Thanks to OWD's BCD collector project, I can verify the following:
https://mdn-bcd-collector.gooborg.com/tests/api/Fence
https://mdn-bcd-collector.gooborg.com/tests/api/HTMLFencedFrameElement
https://mdn-bcd-collector.gooborg.com/tests/api/Navigator/deprecatedReplaceInURN
https://mdn-bcd-collector.gooborg.com/tests/api/Window/fence

However, in `FencedFrameConfig`, there are a few properties that seem unsupported for the moment:
https://mdn-bcd-collector.gooborg.com/tests/api/FencedFrameConfig
```
api.FencedFrameConfig.containerHeight (Window exposure):  No Support
api.FencedFrameConfig.containerWidth (Window exposure):  No Support
api.FencedFrameConfig.contentHeight (Window exposure):  No Support
api.FencedFrameConfig.contentWidth (Window exposure):  No Support
```

I also don't see them in the IDL:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/fenced_frame/fenced_frame_config.idl

Can you remove these from this PR or do you have other information?

**Comment by chrisdavidmills** [comment]
> Can you remove these from this PR or do you have other information?

@Elchi3 I suspect this isn't that simple and is to do with the "opaque identifiers" concept around fenced frames (look at the [IDL listing](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/fenced_frame/fenced_frame_config.idl), lines 12-15). Saying that, I don't fully understand this, so I will ask the Chrome engineers for some help here.

**Comment by Elchi3** [comment]
@chrisdavidmills Any news? This PR could land if you remove the 4 properties.

**Comment by chrisdavidmills** [comment]
> @chrisdavidmills Any news? This PR could land if you remove the 4 properties.

I finally managed to talk to someone who can help me track down this information; I should have a definite answer about these properties very soon.

**Comment by queengooborg** [comment]
Welp...it looks like this is another set of APIs that doesn't seem to be available in BrowserStack.  The collector results are reporting these APIs as unsupported.

**Pull request merged**